routes/ci.js

```diff
@@ -85,17 +114,92 @@ router.get('/:buildItemId/:accessToken', function(req, res, next) {
   res.json( { message: 'Request accepted' } );
 });
 
+
+/**
+ * Receive a user data for specific REST token.
+ *
+ * @param accessToken
+ *  Drupal restful private user token
+ *
+ * @returns {Promise}
+ *  HTTP Response in promise wrapper.
+ */
+var getUser = function(accessToken) {
+  var options = {
+    url: backendUrl + '/api/me/',
+    qs: {
+      access_token: accessToken,
+      fields: 'id,label,github_access_token',
+      github_access_token: true
+    }
+  };
+
+  return request(options);
+};
+
+/**
+ * Get CI Build data.
+ *
+ * @param buildId
+ *  The build ID.
+ * @param accessToken
+ *  Drupal restful private user token
+ *
+ * @returns {Promise}
+ *  HTTP Response in promise wrapper.
+ */
+var getBuild = function(buildId, accessToken) {
+  var options = {
+    url: backendUrl + '/api/ci-builds/' + buildId,
+    qs: {
+      access_token: accessToken,
+      fields: 'id,label,git_branch,repository,private_key'
+    }
+  };
+
+  return request(options);
+};
+
+/**
+ * Get shoov configuration file from user repository.
+ *
+ * @param userName
+ *  The owner username of repository.
+ * @param repositoryName
+ *  The repository name.
+ * @param accessToken
+ *  Drupal restful private user token
+ *
+ * @returns {Promise}
+ *  HTTP response in promise wrapper.
+ */
+var getShoovConfig = function(userName, repositoryName, accessToken) {
```

Comment:
need to take care of the case that `.shoov.yml` doesn't exist yet-- so we can actually check for it earlier, and if no `.shoov.yml` return early.